Documentation/specs/corehost.md

```diff
@@ -0,0 +1,112 @@
+# corehost runtime/assembly resolution
+
+The shared host locates assemblies and native libraries using a combination of: Servicing Index, Files in the application folder (aka "app-local") and files from package caches.
+
+## Definitions and Formats
+
+### Servicing Index
+
+The servicing index is loaded when the `DOTNET_SERVICING` environment variable is non-empty. When this variable is non-empty, it points to a directory that will be the **Servicing Root**.
+
+An index file is located at the path defined by `$DOTNET_SERVICING/servicing_index.txt`. In this file are a series of lines of one of the following formats:
+
+```
+# Lines starting with a '#' and blank lines are ignored.
+
+# Identifies an asset from a NuGet Package that has been serviced.
+package|[Package ID]|[Package Version]|[Original Asset Relative Path]=[New Asset Path, relative to Servicing Root]
+package|System.Threading.Thread|1.2.3.4|lib/dotnet5.4/System.Threading.Thread.dll=patches/abc123/System.Threading.Thread.dll
+
+# TBD: Host/Runtime servicing entries.
+```
+
+Paths in this file are **always** specified using `/`, even on Windows. They must be converted to platform-specific directory separators.
+
+This index is loaded when needed during the Resolution Process (see below).
+
+### Dependencies File
```

Comment:
@anurse @schellap Can you please elaborate on the answer? When is .deps not available since in those scenarios, we cannot use servicing or shared FX? The premise was that we will have .deps file in all the scenarios.